src/main/java/com/google/devtools/build/lib/bazel/repository/starlark/StarlarkBaseExternalContext.java

```diff
@@ -223,6 +224,20 @@ private static ImmutableMap<URI, Map<String, List<String>>> getAuthHeaders(
     return res;
   }
 
+  private static ImmutableMap<String, List<String>> getHeaderContents(Dict<?, ?> x, String what)
+  throws EvalException {
+    // Dict.cast returns Dict<String, raw Dict>.
+    @SuppressWarnings({"unchecked", "rawtypes"})
+    Dict<String, List> headersUnchecked = (Dict) Dict.cast(x, String.class, List.class, what);
```

Comment:
If you use `Sequence` instead of `List` here, you can use https://cs.opensource.google/bazel/bazel/+/master:src/main/java/net/starlark/java/eval/Sequence.java;l=36;bpv=1;bpt=1 to cast the inner lists into lists of `String`s without riding Java crashes when users specify incorrect types.